### PR TITLE
Referee emailer

### PR DESCRIPTION
**Pull request by fofr**
### Context

- Add mailer and template for requesting references from referees.

Personalise email by:
- including candidate name
- generating a link to a prefilled Google form

Referee name is missing at the moment, it's being added to the reference model in a separate PR by @aldavidson.

It’s not tied to any application state changes, I think for now we’d want to manually trigger them. But this means we don’t need to be copying and pasting long Google form URLs into Zendesk, plus other fields, and there’s less chance for human error (like confusing referee name and candidate name).

Based on the existing candidate_mailer.

https://trello.com/c/JcBL1Ejh/1249-build-system-to-request-and-receive-references

### CODE REVIEW

File: app/mailers/referee_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def google_form_url_for(candidate_name, reference)
+    t('reference_request.google_form_url') + '?' +
+      {
+        t('reference_request.email_entry') => reference.email_address,
+        t('reference_request.reference_id_entry') => reference.id,
+        t('reference_request.candidate_name_entry') => candidate_name,
+        t('reference_request.referee_name_entry') => 'TODO: Referee name',
+      }.to_query.gsub('+', '%20')
```
Comment (duncanjbrown) [inline app/mailers/referee_mailer.rb]: Any reason not to use `URI.escape`? https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI/Escape.html#method-i-encode

File: app/views/referee_mailer/reference_request_email.text.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+If we don’t receive your reference within 10 working days from the date of this email, we’ll ask <%= @candidate_name %> to suggest another referee.
+
+# Privacy notice
+
+<%= @candidate_name %> put us in touch with you so that you can provide a reference for them.
+
+The Department for Education and the teacher training provider(s) the candidate has applied to have a ‘legitimate interest’ under data protection legislation to hold your data for the period needed to process the candidate’s application.
+
+We’ll only use your data for this purpose and will delete it when no longer needed for this purpose.
```
Comment (duncanjbrown) [inline app/views/referee_mailer/reference_request_email.text.erb]: Maybe the second "for this purpose" is redundant?

File: app/views/referee_mailer/reference_request_email.text.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+We currently use secure G Suite tools to process reference data. Your data will only be accessed by the staff who need it to process reference data.
```
Comment (duncanjbrown) [inline app/views/referee_mailer/reference_request_email.text.erb]: Is "G suite" a bit jargony? Do we need that first sentence?

Comment (fofr) [reply app/views/referee_mailer/reference_request_email.text.erb]: Agree. This is based on a first draft by Emma F and I'll feedback.

File: app/mailers/referee_mailer.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (fofr) [reply app/mailers/referee_mailer.rb]: Done in a8cc2ab

File: app/views/referee_mailer/reference_request_email.text.erb (hunk 3 above)
Comment (fofr) [reply app/views/referee_mailer/reference_request_email.text.erb]: Section removed in ccc8ad282c5e3b3ae2f9c11e14953929d66e7381

**Pull request merged**